src/main/java/org/terasology/potions/system/DrinkPotionAuthoritySystem.java

```diff
@@ -43,23 +49,58 @@
 import org.terasology.potions.events.BeforeDrinkPotionEvent;
 import org.terasology.potions.events.DrinkPotionEvent;
 import org.terasology.logic.common.ActivateEvent;
+import org.terasology.registry.CoreRegistry;
 import org.terasology.registry.In;
 import org.terasology.utilities.Assets;
 
+/**
+ * This authority system handles potion consumption by various entities.
+ */
 @RegisterSystem(value = RegisterMode.AUTHORITY)
 public class DrinkPotionAuthoritySystem extends BaseComponentSystem {
+    /**
+     * Used for playing drinking sounds.
+     */
     @In
     private AudioManager audioManager;
 
+    /**
+     * This Context is necessary for all the AlterationEffects due to timing and use of the DelayManager.
+     */
     @In
     private Context context;
 
+    /**
+     * Check the potion drink event by sending out a consumable event so that other systems can either modify the
+     * effect parameters or cancel (consume) the event outright. Following that, if not consumed, apply the effect to
+     * the instigator.
+     * This calls the {@link #checkDrink(EntityRef, EntityRef, PotionComponent, HerbEffect, PotionEffect, String)}}
+     * method but leaves the id field empty.
+     *
+     * @param instigator    The entity who is trying to consume this potion.
+     * @param item          A reference to the potion item entity which has the potion effect.
+     * @param p             The PotionComponent of the potion item.
```

Comment:
although well documented, please remove the abbreviated variable names and change them to something more clear